### PR TITLE
Issue #7486: Xpath implementation: relationship between getLocalPart and getStringValue

### PR DESCRIPTION
**Pull request by timurt**
https://github.com/checkstyle/checkstyle/issues/7486

### CODE REVIEW

**Comment by romani** [comment]
@timurt , please look at question in issue ... provide reason why it is better to throw exception.

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java [hunk 1]
```diff
@@ -157,6 +157,15 @@ public final TreeInfo getTreeInfo() {
         return treeInfo;
     }
 
+    /**
+     * Returns string value.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java]: please fix javadoc

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java [hunk 2]
```diff
@@ -157,6 +157,15 @@ public final TreeInfo getTreeInfo() {
...
+     * @return string value
+     */
+    @Override
+    public String getStringValue() {
+        throw throwUnsupportedOperationException();
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java]: `throw throw`

method should be named as "createUnsupportedOperationException"

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java (hunk 1 above)
Comment (timurt) [reply src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java (hunk 2 above)
Comment (timurt) [reply src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java]: done

**Pull request merged**